src/data/abilities/ab-attrs/force-switch-out-immunity-ab-attr.ts

```diff
@@ -1,16 +1,15 @@
 import { AbAttr } from "#abilities/ab-attr";
 import { AbAttrFlag } from "#enums/ab-attr-flag";
 import type { Pokemon } from "#field/pokemon";
-import type { BooleanHolder } from "#utils/common-utils";
+import type { ValueHolder } from "#utils/common-utils";
 
 export class ForceSwitchOutImmunityAbAttr extends AbAttr {
-  constructor(showAbility: boolean = true, showAbilityInstant: boolean = false) {
-    super(showAbility, showAbilityInstant);
+  constructor() {
+    super();
```

Comment:
Suction Cups should show a flyout: https://www.youtube.com/watch?v=lW6p9GCrRS4
and Guard Dog shouldn't (for some reason...): https://www.youtube.com/watch?v=rggthHKGgTM
```suggestion
  constructor(showAbility: boolean) {
    super(showAbility);
```

Reply:
This would require an almost full rewrite of the attribute because the activation message is currently handled externally wherever this is applied. It used to use an async flyout alongside the external messages to show the ability.

Also, I'd rather just display Guard Dog. Both abilities currently use a generic "can't be switched out" message when activated. Should they use different messages?

Reply:
Having both display the flyout is fine, it makes more sense that way (Gamefreak please...).
Suction Cups has its own message, but Guard Dog can continue to use a generic one.

Reply:
Just slap a comment and call it a day. It's not a major thing and a lot of Pokemon fangames have qol ability flyouts even when it's not technically canon 🤷

Reply:
Added a TODO to the attribute